controllers/storagecluster/storageconsumer.go

```diff
@@ -76,6 +76,14 @@ func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageClus
 			util.AddAnnotation(storageConsumer, defaults.KeyRotationEnableAnnotation, val)
 		}
 
+		if storageCluster.Spec.ManagedResources.CephNonResilientPools.Enable {
+			// add a annotation to the storageconsumer that has the topology key for non resilient pools
+			topologyKey := storageCluster.Status.FailureDomainKey
+			util.AddAnnotation(storageConsumer, util.AnnotationNonResilientPoolsTopologyKey, topologyKey)
+		} else {
+			util.AddAnnotation(storageConsumer, util.AnnotationNonResilientPoolsTopologyKey, "")
```

Comment:
There is no point in having the annotation in this case, please remove the annotation completely

Reply:
Added this check for `disabling the replica-1`.